app/assets/javascripts/discourse/app/components/footer-nav.gjs

```diff
@@ -0,0 +1,179 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { modifier as modifierFn } from "ember-modifier";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import htmlClass from "discourse/helpers/html-class";
+import { postRNWebviewMessage } from "discourse/lib/utilities";
+import { SCROLLED_UP, UNSCROLLED } from "discourse/services/scroll-direction";
+import not from "truth-helpers/helpers/not";
+
+class FooterNav extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service scrollDirection;
+  @service composer;
+  @service modal;
+
+  @tracked shouldToggleMobileFooter = false;
+  @tracked canGoBack = false;
+  @tracked canGoForward = false;
+
+  currentRouteIndex = 0;
+  routeHistory = [];
+  backForwardClicked = false;
+
+  registerAppEvents = modifierFn(() => {
+    this.appEvents.on("page:changed", this, "_routeChanged");
+
+    return () => {
+      this.appEvents.off("page:changed", this, "_routeChanged");
+    };
+  });
+
+  @action
+  _routeChanged(route) {
+    // only update route history if not using back/forward nav
+    if (this.backForwardClicked) {
+      this.backForwardClicked = null;
+      return;
+    }
+
+    this.routeHistory.push(route.url);
```

Comment:
Seems like routeHistory will grow forever, and would therefore be a memory leak? Should we prune it to 100 values, or similar?